### PR TITLE
added method ROK4a

### PR DESCRIPTION
**Pull request by cwittens**
## Checklist

- [x] Appropriate tests were added
- [ ] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [ ] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [ ] Any new documentation only uses public API
  
## Additional context

added method ROK4a from [Rosenbrock--Krylov Methods for Large Systems of Differential Equations](https://doi.org/10.1137/130923336)
I dont think OrdinaryDiffEq currently utilizes Krylov Solvers for solving the Jacobian, but I need the `_transformtab(Alpha,Gamma,B,Bhat)` in `function ROK4aTableau()` for a different project, so I thought I might as well implement the method, if I already added the Tableau.

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
> I dont think OrdinaryDiffEq currently utilizes Krylov Solvers for solving the Jacobian

That's a choice. Just change `linsolve = KrylovJL_GMRES()` and it'll use a Krylov method. The tableau and the way the linear equations are solved are not 1-1.

**Comment by cwittens** [comment]
Ohh I didn't know that.
What do you mean with _The tableau and the way the linear equations are solved are not 1-1._ ?

**Comment by ChrisRackauckas** [comment]
Every tableau can be used with every linear solver, and every linear solver can be used with every tableau.

**Comment by ChrisRackauckas** [comment]
What's the next steps here?

**Comment by cwittens** [comment]
are the failed checks from this method? I thought they were general. Otherwise I will try and look for a fix.

The "different project" I was talking about is something in Fortran where I needed the coefficients in the `a,C,b,btilde,d,c `format and not `Alpha,Gamma,B,Bhat` as given in the paper, which is why I used `_transformtab(Alpha,Gamma,B,Bhat)`. 
And after already implementing the `Alpha,Gamma,B,Bhat` coefficients, i thought I made as well add this as a new method, but I didnt plan to do more with it in julia/ordinarydiffeq.

File: src/OrdinaryDiffEq.jl [hunk 1]
```diff
@@ -419,7 +419,7 @@ export Rosenbrock23, Rosenbrock32, RosShamp4, Veldd4, Velds4, GRK4T, GRK4A,
        Ros4LStab, ROS3P, Rodas3, Rodas23W, Rodas3P, Rodas4, Rodas42, Rodas4P, Rodas4P2,
        Rodas5, Rodas5P, Rodas5Pe, Rodas5Pr,
        RosenbrockW6S4OS, ROS34PW1a, ROS34PW1b, ROS34PW2, ROS34PW3, ROS34PRw, ROS3PRL,
-       ROS3PRL2,
+       ROS3PRL2, ROK4a
```
Comment (ChrisRackauckas) [inline src/OrdinaryDiffEq.jl:422]:
```suggestion
       ROS3PRL2, ROK4a,
```

**Comment by ChrisRackauckas** [comment]
The failed checks were from the method, but it was just a typo so I handled it. 

Thanks! Did you benchmark it on a few problems to see how it works in practice?

**Comment by cwittens** [comment]
Thanks.
Nop, I only checked if it works but did not benchmark it in any way.

**Pull request merged**